nrf_rpc/include/nrf_rpc_errors.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef NRF_RPC_ERRORS_H_
+#define NRF_RPC_ERRORS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup nrf_rpc_errors Error codes for nRF RPC
+ * @{
+ * @ingroup nrf_rpc
+ *
+ * @brief Error codes for nRF RPC.
+ */
+
+enum nrf_rpc_error_code {
```

Comment:
Have you considered reusing errno or NRFX error codes? We should avoid that too different error formats for the libs in nrfxlib

Reply:
We were considering errno, but the code does not match our ceses and the other problem is portability (e.g. in FreeRTOS). I agree that we should avoid using different error formats, so I will reconsider that and I also look at the NRFX error codes.

Reply:
Ok, so I reconsidered the error codes. I can suggest following solution. Do you @rugeGerritsen and other reviewers agree to that?

Error values used by nRF RPC can be reduced to:
```
NRF_RPC_ERR_INTERNAL
NRF_RPC_ERR_INVALID_PARAM
NRF_RPC_ERR_NO_MEM
NRF_RPC_ERR_NOT_SUPPORTED
NRF_RPC_ERR_REMOTE
```

They will map nicely to errno:
```
NRF_RPC_ERR_INTERNAL		-EIO
NRF_RPC_ERR_INVALID_PARAM	-EINVAL
NRF_RPC_ERR_NO_MEM		-ENOMEM
NRF_RPC_ERR_NOT_SUPPORTED	-ENOSYS
NRF_RPC_ERR_REMOTE		-EPROTO
```

To support targets without errno we can remove `nrf_rpc_errors.h` and add following lines into `nrf_rpc_common.h`:
```
#ifndef EIO
#define EIO 5
#endif
#ifndef EINVAL
#define EINVAL 22
#endif
#ifndef ENOMEM
#define ENOMEM 12
#endif
#ifndef EPROTO
#define EPROTO 71
#endif
#ifndef ENOSYS
#define ENOSYS 88
#endif
```

Or add above lines to `nrf_rpc_errors.h` removing old error codes.

Reply:
Current solution is to use part of errno codes and define them in `nrf_rpc_common.h` if not defined yet.

```
#ifndef ENOENT
#define ENOENT 2
#endif
#ifndef EIO
#define EIO 5
#endif
#ifndef ENOMEM
#define ENOMEM 12
#endif
#ifndef EFAULT
#define EFAULT 14
#endif
#ifndef EINVAL
#define EINVAL 22
#endif
#ifndef EBADMSG
#define EBADMSG 77
#endif
```